dataactvalidator/app.py

```diff
@@ -39,22 +42,39 @@ def before_request():
     def handle_response_exception(error):
         """Handle exceptions explicitly raised during validation."""
         logger.error(str(error))
+        job_id = g.get('job_id', None)
+        if job_id:
+            sess = GlobalDB.db().session
+            job = sess.query(Job).filter(Job.job_id == job_id).one_or_none()
+            if job:
+                if job.filename is not None:
+                    # insert file-level error info to the database
+                    writeFileError(job_id, job.filename, error.errorType, error.extraInfo)
+                if error.errorType != ValidationError.jobError:
+                    # job pass prerequisites for validation, but an error
+                    # happened somewhere. mark job as 'invalid'
+                    # next 2 lines are very temporary, until the job interface refactor is done
+                    mark_job_status(job_id, 'invalid')
         return JsonResponse.error(error, error.status)
 
     @app.errorhandler(Exception)
     def handle_validation_exception(error):
         """Handle uncaught exceptions in validation process."""
-        job_id = request.json.get('job_id')
+        logger.error(str(error))
+        job_id = g.get('job_id', None)
 
-        # if request had a job id, set job to failed status
+        # set job to failed status; if job has an associated file
+        # name, insert file-level error info to the database
         if job_id:
             sess = GlobalDB.db().session
-            job = sess.query(Job).filter(Job.job_id == job_id).one()
-            job.status_id = JOB_STATUS_DICT['failed']
-            sess.commit()
+            job = sess.query(Job).filter(Job.job_id == job_id).one_or_none()
+            if job:
+                if job.filename is not None:
+                    writeFileError(job_id, job.filename, ValidationError.unknownError)
+                # next 2 lines are very temporary, until the job interface refactor is done
+                jobDb = JobHandler()
+                mark_job_status(job_id, 'failed')
```

Comment:
We might need to be more careful about corner cases in this function as it's basically a top-level exception handler. Could any of these functions raise an exception that we need to handle immediately?

Also, looks like you don't need `JobHandler` anymore.